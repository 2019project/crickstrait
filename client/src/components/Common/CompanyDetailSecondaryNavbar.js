import React, { Component } from "react";
import {
  getCompanyDetailById,
  getVolatility
} from "../../actions/CompanyDetail";
import { connect } from "react-redux";
import companylogo from "./stockslogo.PNG";

// importing css file
import "../../styles/CompanyDetailSecondaryNavbar.css";
import { Link } from "react-router-dom";

export class CompanyDetailSecondaryNavbar extends Component {
  componentDidMount() {
    this.props.getVolatility("AAPL");
  }
  render() {
    return (
      <div>
        <div id="parent">
          {this.props.company ? (
            <div>
              {this.props.company.map(coms => (
                <>
                  <div id="company-details-tab-container">
                    <>
                      <div id="secondary-navbar-img-name-container">
                        <div id="secondary-navbar-img">
                          <img
                            id="stocks_img"
                            src={
                              coms.image == null
                                ? companylogo
                                : "data:image/jpeg;base64," + coms.image
                            }
                          />
                        </div>

                        <div id="secondary-navbar-ticker_name">
                          <p>{coms.ticker_name}</p>
                          <p id="CompanyDetailsIndexClose">
                            {coms.last_share_price}

                            <sub id="secondary-navbar-sub">USD</sub>
                          </p>
                          <p id="last_date">
                            CLOSED PRICE (
                            {new Date(coms.share_date).toLocaleDateString(
                              "en-IN",
                              {
                                month: "short",
                                day: "2-digit",
                                year: "numeric"
                              }
                            )}
                            )
                          </p>
                        </div>
                        <div />
                      </div>
                      <ul id="company-detail-content-container">
                        <Link
                          className="company-link"
                          // PASSING TO COMPANY DETAIL PAGE WITH THE ID WHICH IS MAPPED FROM THE REDUCER
                          to={{
                            pathname: "/companydetail/" + coms.id
                          }}
                        >
                          <span
                            id="company-detail-overview-click"
                            // onClick={() => {
                            //   // changing the values of the state while clicking, the clicked component will be set to true others will be false
                            //   // this.setState({
                            //   //   overview: true,
                            //   //   financial: false,
                            //   //   analysis: false,
                            //   // });
                            //   console.log("Overview clicked");
                            // }}
                          >
                            <li
                              id="comapany-detail-overview-li"
                              // different css properties based on the value of the state
                              className={
                                this.props.selected === "overview"
                                  ? "options-selected-li"
                                  : "options-li"
                              }
                            >
                              Overview
                            </li>
                          </span>
                        </Link>
                        {/* Financials  */}
                        {/* <div id="company-detail-financial"> */}

                        <div id="company-detail-financial">
                          <Link
                            className="company-link"
                            to={{
                              pathname: "/financial/" + coms.id
                            }}
                          >
                            <span
                              id="company-financial-click"
                              // onClick={() => {
                              //   this.setState({
                              //     overview: false,
                              //     financial: true,
                              //     analysis: false,
                              //   });
                              // }}
                            >
                              <li
                                id="company-detail-financial-li"
                                className={
                                  this.props.selected === "financial"
                                    ? "options-selected-li"
                                    : "options-li"
                                }
                              >
                                Financials
                              </li>
                            </span>
                          </Link>
                        </div>

                        <div id="company-detail-analysis">
                          <Link
                            className="company-link"
                            to={{
                              pathname: "/analysis/" + coms.id
                            }}
                          >
                            <span
                              id="company-analysis-click"
                              // onClick={() => {
                              //   this.setState({
                              //     overview: false,
                              //     financial: false,
                              //     analysis: true
                              //   });
                              // }}
                            >
                              <li
                                id="company-detail-analysis-li"
                                className={
                                  this.props.selected === "analysis"
                                    ? "options-selected-li"
                                    : "options-li"
                                }
                              >
                                Analysis
                              </li>
                            </span>
                          </Link>
                        </div>
                      </ul>
                      <div id="secondary-nav-values">
                        <div id="voltality">
                          <p> {this.props.voltality} </p>
                          <div id="voltality_tooltip">
                            <div id="kk">VOLATILITY </div>
                            <div id="abc">
                              <i class="fa fa-question-circle" />
                              <span id="volatility">
                                Volatility is a statistical measure of the
                                dispersion of returns for a given security or
                                market index. In most cases, the higher the
                                volatility, the riskier the security. Volatility
                                is often measured as either the standard
                                deviation or variance between returns from that
                                same security or market index.
                              </span>
                            </div>
                          </div>
                        </div>
                        {/* ------------------------------------------------------------------ */}
                        <div id="voltality">
                          <p id="market_cap"> {coms.last_market_cap} </p>
                          <div id="kk">MARKET CAP</div>
                        </div>
                        {/* --------------------------------------------------------------------------- */}
                      </div>
                    </>

                    <div id="bla">
                      {localStorage.getItem("token") ? (
                        <div id="secondary-navbar-bla">
                          {/* second grid of secondaru navbar */}
                          <button
                            id="downloadButton"
                            onClick={async () => {
                              fetch(
                                "http://localhost:2001/api/companydetail/indicatorsdata/" +
                                  coms.ticker_name
                              ).then(response => {
                                response.blob().then(blob => {
                                  let url = window.URL.createObjectURL(blob);
                                  let a = document.createElement("a");
                                  a.href = url;
                                  a.download = coms.ticker_name + ".csv";
                                  a.click();
                                });
                                //window.location.href = response.url;
                              });
                              // this.props.downloadOHLC("AAPL");
                            }}
                          >
                            <i class="fa fa-download" /> Download
                          </button>
                        </div>
                      ) : (
                        <div style={{ display: "none" }}></div>
                      )}
                    </div>
                  </div>
                  <></>
                </>
              ))}
            </div>
          ) : (
            <p />
          )}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  company: state.CompanyDetailReducer.company,
  voltality: state.CompanyDetailReducer.voltality
});
export default connect(mapStateToProps, {
  getCompanyDetailById,
  getVolatility
})(CompanyDetailSecondaryNavbar);
