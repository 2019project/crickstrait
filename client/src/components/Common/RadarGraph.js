import React, { Component } from "react";
import { Polar } from "react-chartjs-2";
import { connect } from "react-redux";

export class RadarGraph extends Component {
  state = {
    marksData: {
      labels: [
        "Market Cap",
        "Net Profit",
        "Share Price",
        "Revenue",
        "Total Assets",
        "EBIT"
      ],
      datasets: [
        {
          fill: true,
          backgroundColor: [
            "#FF6384",
            "#4BC0C0",
            "#FFCE56",
            "#E7E9ED",
            "#36A2EB",
            "#2c3e50"
          ],

          data: this.props.data
        }
      ]
    }
  };
  render() {
    return (
      <div>
        <Polar
          options={{
            legend: false,
            labels: false,
            responsive: true
          }}
          data={this.state.marksData}
        />
      </div>
    );
  }
}
const mapStateToProps = state => ({});
export default connect(mapStateToProps, {})(RadarGraph);
