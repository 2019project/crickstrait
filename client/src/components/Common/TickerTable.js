import React, { Component } from "react";
import "../../styles/TickerTable.css";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

export class TickerTable extends Component {
  render() {
    console.log(this.props.isIndex);
    console.log(this.props.tableHeaders);
    return (
      <div className="table__container">
        <table className="ticker__table">
          <thead className="ticker__tableHeader">
            <tr>
              {this.props.tableHeaders.map((data, index) => (
                <th>{data}</th>
              ))}
            </tr>
          </thead>
          {console.log("tabledata", typeof this.props.tableData)}
          {this.props.tableData.map((current, index) => (
            <tr>
              {this.props.isIndex == true ? (
                <Link
                  to={{ pathname: "/indexProfile/" + current.ticker_id }}
                  style={{ textDecoration: "none" }}
                >
                  <td>{current.ticker_name ? current.ticker_name : null}</td>
                </Link>
              ) : (
                <Link
                  to={{ pathname: "/companydetail/" + current.ticker_id }}
                  style={{ textDecoration: "none" }}
                >
                  <td>{current.ticker_name ? current.ticker_name : null}</td>
                </Link>
              )}
              {current.tickerValues
                ? Object.keys(current.tickerValues).map((keyName, index) => (
                    <>
                      {current.tickerValues[keyName].substring(0, 1) == "-" ? (
                        <td style={{ color: "#ff4d4d" }}>
                          {current.tickerValues[keyName]}
                        </td>
                      ) : current.tickerValues[keyName].substring(0, 1) ==
                        "+" ? (
                        <td style={{ color: "#27ae60" }}>
                          {current.tickerValues[keyName]}
                        </td>
                      ) : (
                        <td>{current.tickerValues[keyName]}</td>
                      )}
                    </>
                  ))
                : null}
            </tr>
          ))}
        </table>
      </div>
    );
  }
}
const mapStateToProps = state => ({});
export default connect(mapStateToProps, {})(TickerTable);
