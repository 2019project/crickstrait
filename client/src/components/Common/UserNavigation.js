import React, { Component } from "react";
import "../../styles/UserNavigation.css";
import wallet from "./wallet.png";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { getWallet } from "../../actions/Users";
import jwt_decode from "jwt-decode";

export class UserNavigation extends Component {
  componentDidMount() {
    if (localStorage.getItem("token")) {
      let email = {
        email: "admin@gmail.com"
      };
      this.props.getWallet(email);
    }
  }

  render() {
    if (localStorage.getItem("token")) {
      var decode = jwt_decode(localStorage.getItem("token"));
    }
    return (
      <div>
        <div id="userNavigationContainer">
          <div id="userNavigation">
            {localStorage.getItem("token") ? (
              <h1>Hi {decode.name} !</h1>
            ) : (
              <h1>Hi User !</h1>
            )}
            <img src={wallet} id="wallet" />{" "}
            <span id="walletPrice">
              ${Number(this.props.wallet).toFixed(2)}
            </span>
            <div id="userNavigationButtonContainer">
              <Link to="/profile">
                <button
                  id={
                    this.props.selected === "profile"
                      ? "buyStocks-active"
                      : "buyStocks"
                  }
                >
                  Buy Stocks
                </button>
              </Link>
              <Link to="/buyStocks">
                <button
                  id={
                    this.props.selected === "buy"
                      ? "myStocks-active"
                      : "myStocks"
                  }
                >
                  My Stocks
                </button>
              </Link>
              <Link to="/userhistory">
                <button
                  id={
                    this.props.selected === "history"
                      ? "historyStocks-active"
                      : "historyStocks"
                  }
                >
                  History
                </button>
              </Link>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  wallet: state.usersReducer.wallet
});

export default connect(mapStateToProps, { getWallet })(UserNavigation);
