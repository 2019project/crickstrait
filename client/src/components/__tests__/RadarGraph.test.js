import React from "react";
import { shallow, mount } from "enzyme";
import { RadarGraph } from "../Common/RadarGraph";

const radarGraph = jest.fn();

const wrapper = shallow(<RadarGraph radarGraph={radarGraph} />);

describe("Testing radar graph Component", () => {
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
