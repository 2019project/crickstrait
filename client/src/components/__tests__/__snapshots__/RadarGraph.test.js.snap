// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing radar graph Component should mount the component 1`] = `
<div>
  <Polar
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": Array [
              "#FF6384",
              "#4BC0C0",
              "#FFCE56",
              "#E7E9ED",
              "#36A2EB",
              "#2c3e50",
            ],
            "data": undefined,
            "fill": true,
          },
        ],
        "labels": Array [
          "Market Cap",
          "Net Profit",
          "Share Price",
          "Revenue",
          "Total Assets",
          "EBIT",
        ],
      }
    }
    options={
      Object {
        "labels": false,
        "legend": false,
        "responsive": true,
      }
    }
  />
</div>
`;
